Source/WebCore/dom/Element.cpp

```diff
@@ -4798,6 +4798,19 @@ bool Element::isWritingSuggestionsEnabled() const
             return false;
     }
 
+    // This is not yet part of the spec, but it improves web-compatibility; if autocomplete
+    // is intentionally off, the site author probably wants writingsuggestions off too.
+
+    for (auto* ancestor = this; ancestor; ancestor = ancestor->parentElementInComposedTree()) {
```

Comment:
Per our smart ptr guidelines
```suggestion
    for (RefPtr ancestor = this; ancestor; ancestor = ancestor->parentElementInComposedTree()) {
```